x-pack/plugins/apm/public/plugin.ts

```diff
@@ -114,6 +133,23 @@ export class ApmPlugin implements Plugin<ApmPluginSetup, ApmPluginStart> {
         return renderApp(coreStart, pluginSetupDeps, params, config);
       },
     });
+
+    core.application.register({
+      id: 'csm',
+      title: 'Client Side Monitoring',
+      order: 8500,
+      category: DEFAULT_APP_CATEGORIES.observability,
+
+      async mount(params: AppMountParameters<unknown>) {
+        // Load application bundle
+        const { renderApp } = await import('./application/csmApp');
+        // Get start services
+        const [coreStart] = await core.getStartServices();
+        await setupLazyStuff(coreStart);
```

Comment:
Parallelisation could be improved (`Promise.all`)

Reply:
improved